### PR TITLE
Data Layer: Track data request meta information

### PR DESCRIPTION
**Pull request by samouri**
exploring a redux-based alternative implementation of https://github.com/Automattic/wp-calypso/pull/17263

### CODE REVIEW

File: client/state/data-layer/reducer.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	return deterministicStringify( action );
+};
+
+const requestStatus = ( state = {}, action ) => {
+	if ( action.type === TRACK_REQUEST ) {
+		const { requestAction } = action;
+
+		return {
+			...state,
+			[ getActionKey( requestAction ) ]: {
```
Comment (dmsnell) [inline client/state/data-layer/reducer.js]:
```js
const keyedReducer = ( identifier, reducer ) => action => {
	const key = 'function' === typeof identifier
		? identifier( action )
		: action[ identifier ];

	…
}
```

eh? eh? :smile:

**Comment by dmsnell** [comment]
I'm less excited about this concept due to how much it feels like the data layer leaking into the larger app. Also it's more dispatches which I wonder if they are needed.

I'll have to think about it more to provide better feedback. Something just doesn't feel right about tracking this here. Thanks for providing this alternative PR though!

**Comment by apeatling** [comment]
@samouri How are you feeling about this exploration? Should we take this further or close it off?

**Comment by samouri** [comment]
@apeatling: it isn't worth investigating further, closing

**Pull request closed without merging**